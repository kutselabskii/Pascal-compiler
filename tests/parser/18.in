program abc;

procedure a(a:integer);
begin
	a := 5;
end

begin
	a(2);
end.