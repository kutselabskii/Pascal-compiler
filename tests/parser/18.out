|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- a	//Procedure_declaration
        |   |   |-- //Arguments list
        |   |   |   |-- a
        |   |   |   |   |__  : integer
        |   |   |__ //Block
        |   |       |-- //Declaration part
        |   |       |__ //Statement part
        |   |           |__ //Statement sequence
        |   |               |-- //Assignment_statement
        |   |               |   |-- a
        |   |               |   |   |__  : integer
        |   |               |   |__ 5
        |__ //Statement part
            |__ //Statement sequence
                |-- a//Procedure statement
                |   |-- 2
